lib/ui/painting/picture.cc

```diff
@@ -16,6 +18,54 @@
 #include "third_party/tonic/logging/dart_invoke.h"
 
 namespace flutter {
+namespace {
+
+sk_sp<SkSurface> MakeSnapshotSurface(const SkISize& picture_size,
+                                     fml::WeakPtr<GrContext> resource_context) {
+  SkImageInfo image_info = SkImageInfo::MakeN32Premul(
+      picture_size.width(), picture_size.height(), SkColorSpace::MakeSRGB());
+  if (resource_context) {
+    return SkSurface::MakeRenderTarget(resource_context.get(),  // context
+                                       SkBudgeted::kNo,         // budgeted
+                                       image_info               // image info
+    );
+  } else {
+    return SkSurface::MakeRaster(image_info);
+  }
+}
+
+sk_sp<SkImage> MakeRasterSnapshot(sk_sp<SkPicture> picture,
```

Comment:
Done.  I'll add an issue once this lands.

Reply:
I'd like to add that #2 was discussed here at length:
[https://github.com/flutter/engine/pull/8835](https://github.com/flutter/engine/pull/8835)

Would it be possible to work #2 in since this section of code is related? Having a GPU-backed image and avoiding the GPU-to-CPU overhead would be beneficial.

Adding @Hixie - I will soon address the concerns brought up there about increased documentation and tests in this area.

Reply:
> I'd like to add that #2 was discussed here at length:
> #8835
> 
> Would it be possible to work #2 in since this section of code is related? Having a GPU-backed image and avoiding the GPU-to-CPU overhead would be beneficial.
> 
> Adding @Hixie - I will soon address the concerns brought up there about increased documentation and tests in this area.

Noted.  I think it's something we should do as well, but I think it should be prioritized against other issues and not serve as an impediment to this change.